### ISSUE TITLE
Image artifact with tf.keras data augmentation

### ISSUE THREAD

**Issue by tjdurant** [tensorflow/tensorflow#44501]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
- TensorFlow installed from (source or binary): RUN pip install --upgrade tensorflow tensorflow-gpu
- TensorFlow version (use command below): v2.3.0-54-gfcc4b966f1 2.3.1
- Python version: Python 3.6
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version: V10.1.243
- GPU model and memory: GeForce GTX 108 / 261MiB / 11178MiB

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with:
1. TF 1.0: `python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"`
2. TF 2.0: `python -c "import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`

**Describe the current behavior**

When using the following code: 
```
data_augmentation = tf.keras.Sequential([
  tf.keras.layers.experimental.preprocessing.RandomFlip('horizontal'),
  tf.keras.layers.experimental.preprocessing.RandomRotation(90),
])
```
I see black lines around the edges of the augmented images. I do not see this when I use plain Keras:

```
datagen = ImageDataGenerator(
    rotation_range=model_params['rotation_range'],  # randomly rotate images in the range (degrees, 0 to 180)  
    horizontal_flip=model_params['horizontal_flip'],  # randomly flip images
    vertical_flip=model_params['vertical_flip'],

)  
```

Also, when I upgrade to tf-nightly it goes away. 

**Describe the expected behavior**
I would expect the black lines to not be there. 

**Standalone code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate
the problem. If possible, please share a link to Colab/Jupyter/any notebook.

https://colab.research.google.com/drive/1nRpmyMrXkV43W82nI2fP2F05JndeGdg0?usp=sharing

**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.

I guess my question is whether or not these black lines are truly being fed into the network when using data augmentation.

**Comment by amahendrakar** [comment]
Was able to reproduce the issue with [TF v2.3](https://colab.research.google.com/gist/amahendrakar/32d1f447194b885ef65cbeb2ece787af/44501.ipynb#scrollTo=boucrfpl-SrX) and as mentioned issue seems to be resolved in the latest [TF-nightly](https://colab.research.google.com/gist/amahendrakar/2c65092d39820454d5f73e58dd6d63dc/44501-tf-nightly.ipynb#scrollTo=boucrfpl-SrX). 

Please find the attached gist. Thanks!

**Comment by ymodak** [comment]
The fix is available in newly released TF 2.4.0-rc0 as well. Hence we can expect it in TF 2.4 final version as well.
Thanks!

**Comment by tjdurant** [comment]
> The fix is available in newly released TF 2.4.0-rc0 as well. Hence we can expect it in TF 2.4 final version as well.
> Thanks!

Thank you @ymodak. And just to clarify, TF 2.4.0 isn't released yet, correct? So should I use tf-nightly in the interm?

**Comment by ymodak** [comment]
That's correct, 2.4.0 is yet to be released. For now you can use 2.4.0-rc0 or latest nightly version (2.5.XXXX)
